frontend/src/components/utils/NavBar/index.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { ReactElement, useState, useEffect } from 'react';
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  makeStyles,
+  Button,
+  IconButton,
+  Drawer,
+  Link,
+  MenuItem,
+} from '@material-ui/core';
+import MenuIcon from '@material-ui/icons/Menu';
+import { Link as RouterLink } from 'react-router-dom';
+import icon from '../../../assets/home-icon.png';
+
+export type navbarButton = {
+  label: string;
+  href: string;
+};
+
+type Props = {
+  readonly headersData: navbarButton[];
+};
+
+const useStyles = makeStyles(() => ({
+  grow: {
+    flexGrow: 1,
+  },
+  header: {
+    backgroundColor: 'white',
+    paddingTop: '3%',
+    paddingLeft: '7%',
+    paddingRight: '7%',
+    boxShadow: 'none',
+  },
+  logo: {
+    fontFamily: 'Work Sans, sans-serif',
+    fontWeight: 500,
+    '@media only screen and (max-width: 320px) ': {
+      fontSize: '1.3em',
+    },
+    color: 'black',
+    textAlign: 'left',
+  },
+  menuButton: {
+    fontFamily: 'Work Sans, sans-serif',
+    size: '18px',
+    color: 'black',
+    fontSize: '1.2rem',
+    textTransform: 'none',
+    marginLeft: '38px',
+  },
+  toolbar: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  drawerContainer: {
+    padding: '20px 30px',
+  },
+  homeImage: {
+    marginBottom: '8px',
+  },
+  menuDrawer: {
+    alignSelf: 'right',
+    marginBottom: '8px',
+  },
+}));
+
+const NavBar = ({ headersData }: Props): ReactElement => {
+  const [state, setState] = useState({
```

Comment:
You should parametrize this state hook using `useState<TypeName>({...})` so that we can't just pass any random type and set the state to that (might have to make a new type with `mobileView: boolean` and `drawerOpen: boolean`)